packages/wdio-browserstack-service/src/Percy/Percy-Handler.ts

```diff
@@ -0,0 +1,176 @@
+                return true
+            }
+        } else if ((args.method as string) === 'DELETE' && (args.endpoint as string) === '/session/:sessionId') {
+            return true
+        }
+        return false
+    }
+
+    async cleanupDeferredScreenshots() {
+        this.isPercyCleanupProcessingUnderway = true
+        for await (const entry of this.percyDeferredScreenshots) {
+            await this.percyAutoCapture(entry.eventName, entry.sessionName)
+        }
+        this.percyDeferredScreenshots = []
+        this.isPercyCleanupProcessingUnderway = false
+    }
+
+    async #sleep(ms: number) {
+        return new Promise(resolve => setTimeout(resolve, ms))
+    }
+
+    async browserBeforeCommand (args: BeforeCommandArgs) {
+        try {
+            if (this.isDOMChangingCommand(args)) {
+                do {
+                    await this.#sleep(1000)
+                } while (this.percyScreenshotInterval)
+                this.percyScreenshotInterval = setInterval(async () => {
+                    if (!this.isPercyCleanupProcessingUnderway) {
+                        clearInterval(this.percyScreenshotInterval)
+                        await this.cleanupDeferredScreenshots()
+                        this.percyScreenshotInterval = null
+                    }
+                }, 1000)
+            }
+        } catch (err: any) {
+            PercyLogger.error(`Error while trying to cleanup deferred screenshots ${err}`)
+        }
+    }
+
+    async browserAfterCommand (args: BeforeCommandArgs & AfterCommandArgs) {
+        try {
+            if (args.endpoint && this._percyAutoCaptureMode) {
+                let eventName = null
+                if ((args.endpoint as string).includes('click') && ['click', 'auto'].includes(this._percyAutoCaptureMode as string)) {
+                    eventName = 'click'
+                } else if ((args.endpoint as string).includes('screenshot') && ['screenshot', 'auto'].includes(this._percyAutoCaptureMode as string)) {
+                    eventName = 'screenshot'
+                } else if ((args.endpoint as string).includes('actions') && ['auto'].includes(this._percyAutoCaptureMode as string)) {
+                    if (args.body && args.body.actions && Array.isArray(args.body.actions) && args.body.actions.length && args.body.actions[0].type === 'key') {
+                        eventName = 'keys'
+                    }
+                } else if ((args.endpoint as string).includes('/session/:sessionId/element') && (args.endpoint as string).includes('value') && ['auto'].includes(this._percyAutoCaptureMode as string)) {
+                    eventName = 'keys'
+                }
+                if (eventName) {
+                    this.deferCapture(this.sessionName as string, eventName)
+                }
+            }
+        } catch (err: any) {
+            PercyLogger.error(`Error while trying to calculate auto capture parameters ${err}`)
+        }
```

Comment:
No reason to wrap function in try/catch block. Also here, let's return early in all cases
```suggestion
        if (!args.endpoint || !this._percyAutoCaptureMode) {
            return
        }

        let eventName = null
        const endpoint = args.endpoint as string
        if (endpoint.includes('click') && ['click', 'auto'].includes(this._percyAutoCaptureMode as string)) {
            eventName = 'click'
        } else if (endpoint.includes('screenshot') && ['screenshot', 'auto'].includes(this._percyAutoCaptureMode as string)) {
            eventName = 'screenshot'
        } else if (endpoint.includes('actions') && this._percyAutoCaptureMode === 'auto') {
            if (args.body && args.body.actions && Array.isArray(args.body.actions) && args.body.actions.length && args.body.actions[0].type === 'key') {
                eventName = 'keys'
            }
        } else if (endpoint.includes('/session/:sessionId/element') && endpoint.includes('value') && this._percyAutoCaptureMode === 'auto') {
            eventName = 'keys'
        }

        if (eventName) {
            this.deferCapture(this.sessionName as string, eventName)
        }
```

Reply:
Made changes to return early + added try catch here so that any error raised here doesn't bubble up & halt test execution for a user.